app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -723,204 +723,265 @@ describe('NetworkController', () => {
+              });
 
-          const supportsEIP1559 = await controller.getEIP1559Compatibility();
+              const isEIP1559Compatible =
+                await controller.getEIP1559Compatibility();
 
-          expect(supportsEIP1559).toBe(false);
+              expect(isEIP1559Compatible).toBe(true);
+            });
+          });
         });
-      });
-    });
 
-    it('does not make multiple requests to eth_getBlockByNumber when called multiple times and the request to eth_getBlockByNumber succeeded the first time', async () => {
-      await withController(async ({ controller }) => {
-        const fakeProvider = buildFakeProvider([
-          {
-            request: {
-              method: 'eth_getBlockByNumber',
-            },
-            response: SUCCESSFUL_ETH_GET_BLOCK_BY_NUMBER_RESPONSE,
-          },
-        ]);
-        const fakeNetworkClient = buildFakeClient(fakeProvider);
-        mockCreateNetworkClient().mockReturnValue(fakeNetworkClient);
-        await withoutCallingGetEIP1559Compatibility({
-          controller,
-          operation: async () => {
-            await controller.initializeProvider();
-          },
+        describe('if the latest block does not have a "baseFeePerGas" property', () => {
+          it('sets the "1559" property to false', async () => {
+            await withController(async ({ controller }) => {
+              setFakeProvider(controller, {
+                stubs: [
+                  {
+                    request: {
+                      method: 'eth_getBlockByNumber',
+                      params: ['latest', false],
+                    },
+                    response: {
+                      result: PRE_1559_BLOCK,
+                    },
+                  },
+                ],
+                stubLookupNetworkWhileSetting: true,
+              });
+
+              await controller.getEIP1559Compatibility();
+
+              expect(
+                controller.store.getState().networkDetails.EIPS[1559],
+              ).toBe(false);
+            });
+          });
+
+          it('returns false', async () => {
+            await withController(async ({ controller }) => {
+              setFakeProvider(controller, {
+                stubs: [
+                  {
+                    request: {
+                      method: 'eth_getBlockByNumber',
+                      params: ['latest', false],
+                    },
+                    response: {
+                      result: PRE_1559_BLOCK,
+                    },
+                  },
+                ],
+                stubLookupNetworkWhileSetting: true,
+              });
+
+              const isEIP1559Compatible =
+                await controller.getEIP1559Compatibility();
+
+              expect(isEIP1559Compatible).toBe(false);
+            });
+          });
         });
 
-        await controller.getEIP1559Compatibility();
-        await controller.getEIP1559Compatibility();
```

Comment:
Q: what was the thought behind removing this test?

Reply:
Good spot. My thought here was that this behavior is already being tested by two other kinds of tests elsewhere:

1. If `controller.store.getState().networkDetails.EIPS[1559]` is undefined, a call to `getEIP1559Compatibility` should change `networkDetails.EIPS[1559]` in state to either true or false (depending on whether the latest block has a `baseFeePerGas` property
2. If `networkDetails.EIPS[1559]` in state is already true or false, a call to `getEIP1559Compatibility` should not make a network request or make any changes to `networkDetails.EIPS[1559]` in state, but should just return that property

If you put these two together, then you get what this test was doing. That said, I can understand that this is more explicit in demonstrating that `getEIP1559Compatibility` essentially memoizes its result, so what do you think?